pkg/providers/nutanix/template_test.go

```diff
@@ -636,6 +636,31 @@ func TestTemplateBuilderEtcdEncryptionKubernetes129(t *testing.T) {
 	}
 }
 
+func TestTemplateBuilderFailureDomains(t *testing.T) {
```

Comment:
pls add similar test case which creates a cluster without fd and then set fd later to verify if cp nodes are moved to the fd. eg should see following <pre>
kubectl get Cluster  c1 -o json | jq '.status.failureDomains'
{
  "fd1": {
    "controlPlane": true
  }
}
</pre>

<pre>
kubectl get Machine md-0-dsqbk-jfqrh-wmlbd  -o json | jq '.spec.failureDomain'
"fd1"
</pre>